app/services/permissions_checker.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class PermissionsChecker
+  def initialize(current_user:, permission_name:, user_id: '', friendly_id: '', record_id: '')
+    @current_user = current_user
+    @permission_name = permission_name
+    @user_id = user_id
+    @friendly_id = friendly_id
+    @record_id = record_id
+  end
+
+  def call
+    return true if RolePermission.joins(:permission)
+                                 .where(role_id: @current_user.role_id)
+                                 .find_by(permission: { name: @permission_name })&.value == 'true'
```

Comment:
I think you can further remove the `where` and use only the `find_by(role_id: ..., permission: { ... })`